### PR TITLE
[IMP]To improve functionality of hotel and hotel_reservation.

### PR DESCRIPTION
**Pull request by KrishnaPrajapati-SerpentCS**

### CODE REVIEW

File: hotel/models/hotel.py [hunk 1]
```diff
@@ -399,11 +399,10 @@ def onchange_dates(self):
...
         @return: Duration and checkout_date
         '''
-        company_obj = self.env['res.company']
         configured_addition_hours = 0
-        company_ids = company_obj.search([])
-        if company_ids.ids:
-            configured_addition_hours = company_ids[0].additional_hours
+        whouse_com_id = self.warehouse_id.company_id
+        if self.warehouse_id and whouse_com_id:
+            configured_addition_hours = whouse_com_id.additional_hours
```
Comment (JayVora-SerpentCS) [inline hotel/models/hotel.py]: This is a wrong code. 
you are using self.warehouse_id.company_id first and then checking the self.warehouse_id condition.

File: hotel/models/hotel.py [hunk 2]
```diff
@@ -418,9 +417,9 @@ def onchange_dates(self):
             else:
                 myduration = dur.days + 1
             if configured_addition_hours > 0:
-                additional_hours = abs((dur.seconds / 60) / 60)
-                if additional_hours >= configured_addition_hours:
-                    myduration += 1
+                additional_hours = abs((dur.seconds / 60))
+                if additional_hours <= abs(configured_addition_hours * 60):
+                    myduration -= 1
         self.duration = myduration
```
Comment (JayVora-SerpentCS) [inline hotel/models/hotel.py:405]: write a comment why this logic is written

File: hotel/models/hotel.py [hunk 3]
```diff
@@ -863,6 +862,11 @@ def on_change_checkout(self):
         -----------------------------------------------------------------
         @param self: object pointer
         '''
+        configured_addition_hours = 0
+        fwc_id = self.folio_id.warehouse_id.company_id
+        if self.folio_id.warehouse_id and fwc_id:
+            configured_addition_hours = fwc_id.additional_hours
```
Comment (JayVora-SerpentCS) [inline hotel/models/hotel.py]: same as above.

File: hotel_reservation/models/hotel_reservation.py [hunk 4]
```diff
@@ -378,20 +380,14 @@ def _create_folio(self):
...
             for line in reservation.reservation_line:
                 for r in line.reserve:
                     folio_lines.append((0, 0, {
                         'checkin_date': checkin_date,
                         'checkout_date': checkout_date,
                         'product_id': r.product_id and r.product_id.id,
                         'name': reservation['reservation_no'],
-                        'product_uom_qty': ((date_a - date_b).days) + 1,
+                        'product_uom_qty': duration,
                         'is_reserved': True}))
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/models/hotel_reservation.py]: is duration variable defined?

File: hotel_reservation/models/hotel_reservation.py [hunk 5]
```diff
@@ -702,16 +710,76 @@ def get_room_summary(self):
...
+                                                reservline_ids = True
+                                            else:
+                                                reservline_ids = False
+                                        else:
+                                            if hr_dur > 0:
+                                                reservline_ids = True
+                                            else:
+                                                reservline_ids = False
+                                    else:
+                                        reservline_ids = False
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/models/hotel_reservation.py:842]: are you sure we need such a huge code?Write comments each block and explain why whats written

**Comment by KrishnaPrajapati-SerpentCS** [comment]
Ok sir

**Comment by JayVora-SerpentCS** [comment]
@krishnaprajapati16894:
- keep only changes which are of old modules.
- Do not add pos_folio here.
- Correct the conflicts.

**Comment by JayVora-SerpentCS** [comment]
Please check what are the conflict files. Adapt your code acc to them. the changes were necessary for index.html.

**Pull request merged**